Add versioned static test domain

skim1102 opened the pull request:
# About this PR
While implementing features on Shavar service as mentioned in https://github.com/mozilla-services/shavar-list-creation/issues/90, in order to ascertain that the versioned list is being used by the client we need a test domain that only exists in the said versioned list. This PR adds a static versioned test domain in the versioned lists with the following naming convention: `<version>.<list name>.dummytracker.org` like `71.0.base-fingerprinting-track-digest256.dummytracker.org`

# Acceptance Criteria
- [ ] For each versioned list, a static test domain in the following format is added: `<version>.<list name>.dummytracker.org` like `71.0.base-fingerprinting-track-digest256.dummytracker.org`

# Practical Test
### Test that the domain is added to the list using log
1. Go to this [line](https://github.com/mozilla-services/shavar-list-creation/pull/103/commits/8c3d243a0acd54dca95a1074927e48ec21414b11#diff-8c5c15d9cbbefc2ec320aa3f2a253290R274)
2. Add an easily identifable print statement with the domain to be printed on the terminal
3. Run `python ./lists2safebrowsing.py` and check that each versioned list has a unique (versioned) static test domain
### Test that the versioned test domain is used by your client
1. In the browser, go to `about:config` and change the `browser.safebrowsing.provider.mozilla.updateURL` and `browser.safebrowsing.provider.mozilla.gethashURL` to use the staging Shavar
2. Stay on the `about:config` and change the `browser.safebrowsing.provider.mozilla.lists` to just `base-fingerprinting-track-digest256`
3. Go to `about:support` and copy the path listed on the `Profile Folder` row
4. On your laptop, go to the path copied in step 3 and go into the `safebrowsing` folder
5. Remove the files starting with `base-fingerprinting-track-digest256` in the `safebrowsing` folder
6. Back in your browser, go to `about:url-classifier` and under the `Provider` section click the `Trigger Update` button to get the latest changes
7. Refresh the `about:url-classifier` page and in the `URL` search box under `Search` section enter the versioned test URL with the following format `<version>-<list_name>.dummytracker.org` (e.g. `https://71-0-base-fingerprinting-track-digest256.dummytracker.org/test_not_blocked.png`)
8. Click the `Start searching` and check that the versioned test domain is found under the `Results` section in the corresponding category (following the example on step 7, the domain should show up in fingerprinting category).

#
Dependent on https://github.com/mozilla-services/shavar/pull/111

englehardt commented:
I think we should move over to a separate subdomain that we use to check versioning. We currently have a lot of test pages that rely on the `<list_name>.dummytracker.org` formatted test domains, so we don't want to remove them from the list. We also can't add both, because Firefox successively removes subdomains when checking if an origin is blocked. That means that `foo.<list_name>.dummytracker.org` will be blocked for all values of `foo` if `<list_name>.dummytracker.org` is on the list. We could using something like `<version>-<list_name>.dummytracker.org`?

I don't see any config `.ini` files that include `version` as an option. How will this be set an maintained in practice? I know you're planning to use versioned branches in the `shavar-prod-lists` repo, but this seems like it requires versioned branches in the `shavar-list-creation-config` repo as well?

skim1102 commented:
I added the versioning config in the [sample `.ini` file](https://github.com/mozilla-services/shavar-list-creation/blob/master/sample_shavar_list_creation.ini#L33) and also added the versioning config in `shavar-list-creation-config` [here](https://github.com/mozilla-services/shavar-list-creation-config/pull/56). In practice to add a versioned list to the S3 via `shavar-list-creation` we need the following configs setup:
- In `shavar-list-creation-config` repo, go to the `.ini` file and under the section (e.g. `tracking-protection-base` add `versioning_needed=true`
- In `shavar-prod-lists` repo add a branch for versioning by using the version as the branch name (e.g. `72.0a1`, `72.0`, etc)
When the `shavar-list-creation` is ran on the Jenkins, this repo will check for configurations in the `shavar-list-creation-config` and in its first run it will use the master branch on `shavar-prod-lists` and add any changes to the S3. Then it will get all the branches from `shavar-prod-lists` and for any branch with a valid version as its name, this `shavar-list-creation` will create versioned lists for every section that has `versioning_needed=true`. The `version` config added [here](https://github.com/mozilla-services/shavar-list-creation/pull/103/files#diff-8c5c15d9cbbefc2ec320aa3f2a253290R490) is set only when versioned lists are generated. So by using the existing settings, the static test domains for versioned lists will be generated every time new file is created for the tracking protection list during the versioned list generation. Since I was depending on the `versioning_needed` config I purposely did not add `version` on the `.ini` file. Wdyt @englehardt?

skim1102 commented on lists2safebrowsing.py:
```diff
@@ -269,9 +269,11 @@ def write_safebrowsing_blocklist(domains, output_name, allow_list, log_file,
     # Add a static test domain to list
+    test_domain = TEST_DOMAIN_TEMPLATE % output_name
+    if version:
+        test_domain = '{0}-{1}'.format(version.replace('.', '-'), test_domain)
```
@englehardt, updated the test domain to use `-` only (e.g. `71-0-base-fingerprinting-track-digest256.dummytracker.org`)

englehardt commented:
Also LGTM. Thanks for the clarification that the `version` config setting is set during list creation. I missed that!
